### PR TITLE
Add device type to device metadata

### PR DESCRIPTION
**Pull request by NouhaManai96**
### What does this PR do?
Add `device_type` to device metadata

### Motivation
Add support for device type in profiles

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/networkdevice/metadata/payload.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ type DeviceMetadata struct {
 	OsVersion      string       `json:"os_version,omitempty"`
 	OsHostname     string       `json:"os_hostname,omitempty"`
 	Integration    string       `json:"integration,omitempty"` // indicates the source of the data SNMP, meraki_api, etc.
+	DeviceType     string       `json:"device_type,omitempty`
```
Comment (AlexandreYang) [inline pkg/networkdevice/metadata/payload.go:71]:
```suggestion
	DeviceType     string       `json:"device_type,omitempty"`
```

**Review by apigirl — COMMENTED** [review]
Release notes editorial review is 👍

**Review by TCheruy — APPROVED** [review]

File: pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go [hunk 2]
```diff
@@ -228,6 +230,31 @@ func getProfileVersion(config *checkconfig.CheckConfig) uint64 {
...
+		"storage",
+		"switch",
+		"ups",
+		"wlc":
+		return deviceType
+	case "":
+		return "other"
+	}
+	log.Debugf("Unsupported device type: %s", deviceType)
+	return "other"
```
Comment (AlexandreYang) [inline pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go:255]: I still think there is no much value of having "other" as default :) 

Why not just keep empty string as default value?

Trying to better understand what are the benefits of have "other" as default value.

Comment (AlexandreYang) [inline pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go:255]: This looks like a kind of validation, shall we handle that in https://github.com/DataDog/datadog-agent/blob/7b793fe7ecef6c1e47043d39573bfd439946540c/pkg/collector/corechecks/snmp/internal/configvalidation/config_validate_enrich.go ?

File: pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go [hunk 3]
```diff
@@ -228,6 +230,31 @@ func getProfileVersion(config *checkconfig.CheckConfig) uint64 {
 	return profileVersion
 }
 
+func getDeviceType(store *metadata.Store) string {
+	deviceType := strings.ToLower(store.GetScalarAsString("device.type"))
```
Comment (AlexandreYang) [inline pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go:234]: I think you need to add the new field to this list: https://github.com/DataDog/datadog-agent/blob/c42154932ea0b31f418bb5e55f7e46f0ef68952f/pkg/collector/corechecks/snmp/internal/checkconfig/config_validate_enrich.go#L15-L28

File: pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go (hunk 2 above)
Comment (AlexandreYang) [reply pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go:255]: Maybe we can return an error if the type is unsupported?

File: pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go (hunk 3 above)
Comment (NouhaManai96) [reply pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go:250]: addressed in [f14ca10](https://github.com/DataDog/datadog-agent/pull/21791/commits/f14ca1045f043dae18f7581bdf1c9b4c863c643f)

File: pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go (hunk 2 above)
Comment (NouhaManai96) [reply pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go]: addressed in [18d6c0b](https://github.com/DataDog/datadog-agent/pull/21791/commits/18d6c0b82c6c8246e6f2001417ee7bff38088740)

**Review by AlexandreYang — APPROVED** [review]
LGTM overall.

Let's try to check if `sd-wan` is a good candidate as device type, but beside that the rest look good.

File: pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go [hunk 4]
```diff
@@ -228,6 +246,19 @@ func getProfileVersion(config *checkconfig.CheckConfig) uint64 {
...
+	if deviceType == "" {
+		return "other"
+	}
+	_, isValidType := supportedDeviceTypes[deviceType]
+	if isValidType {
+		return deviceType
+	}
```
Comment (AlexandreYang) [inline pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go:257]:
```suggestion
	if supportedDeviceTypes[deviceType] {
		return deviceType
	}
```
Could be simplified like this ☝️ ?

**Review by jhgilbert — APPROVED** [review]

**Pull request merged**